### PR TITLE
[ONNX] Remove special handling of torchvision.ops imports in onnx export

### LINKED ISSUE #141568: ONNX export (dynamo) not working with torchvision ops
### 🐛 Describe the bug

Torchvision ops are not loaded properly,
```python
import torchvision
import torch

class M(torch.nn.Module):
	def __init__(self):
		super().__init__()
		pass

	def forward(self, x, count):
		out = torchvision.ops.batched_nms(x[0], x[1], x[2], x[3])
		return out

model = M()
torch.manual_seed(1)
args = (
	(
		torch.rand(20, 4, dtype=torch.float),
		torch.rand(20, dtype=torch.float),
		torch.randint(0, 2, (20,), dtype=torch.float),
		0,
	),
	3,
)
torch.onnx.export(model, args, dynamo=True)

#torch.onnx.dynamo_export(model, *args)
```
Gives error:

```
<class 'torch.onnx._internal.exporter._errors.DispatchError'>: No ONNX function found for <OpOverload(op='torchvision.nms', overload='default')>. Failure message: No decompositions registered for the real-valued input
```

---

The `torch.onnx.dynamo_export(model, *args)` works fine. As a side note, what is (or will be) the preferred API, the `torch.onnx.export(model, args, dynamo=True)` or the `torch.onnx.dynamo_export(model, *args)`. 

### Versions

<details>
<summary>Versions and HW</summary>
PyTorch version: 2.6.0.dev20241118+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.1 LTS (x86_64)
GCC version: (Ubuntu 13.2.0-23ubuntu4) 13.2.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] (64-bit runtime)
Python platform: Linux-6.8.0-49-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1050 Ti
Nvidia driver version: 560.35.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        36 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               4
On-line CPU(s) list:                  0-3
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i5-2400 CPU @ 3.10GHz
CPU family:                           6
Model:                                42
Thread(s) per core:                   1
Core(s) per socket:                   4
Socket(s):                            1
Stepping:                             7
CPU(s) scaling MHz:                   94%
CPU max MHz:                          3400.0000
CPU min MHz:                          1600.0000
BogoMIPS:                             6185.61
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx lahf_lm epb pti ssbd ibrs ibpb stibp tpr_shadow flexpriority ept vpid xsaveopt dtherm ida arat pln pts vnmi md_clear flush_l1d
Virtualization:                       VT-x
L1d cache:                            128 KiB (4 instances)
L1i cache:                            128 KiB (4 instances)
L2 cache:                             1 MiB (4 instances)
L3 cache:                             6 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-3
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Mitigation; PTE Inversion; VMX conditional cache flushes, SMT disabled
Vulnerability Mds:                    Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Meltdown:               Mitigation; PTI
Vulnerability Mmio stale data:        Unknown: No mitigations
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.1.2
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-cusparselt-cu12==0.6.2
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] onnx==1.17.0
[pip3] onnxscript==0.1.0.dev20241112
[pip3] pytorch-triton==3.1.0+cf34004b8a
[pip3] torch==2.6.0.dev20241118+cu124
[pip3] torchaudio==2.5.0.dev20241118+cu124
[pip3] torchvision==0.20.0.dev20241118+cu124
[conda] Could not collect
</details>

### PR DESCRIPTION
**Pull request by Bludator**
Fixes #141568

### CODE REVIEW

**Review by justinchuby — DISMISSED** [review]
This will remove support for the torch vision ops, right? Is that intended?

**Comment by justinchuby** [comment]
Ok sorry. I think what should be done instead is to remove "ops." from the get attr line

File: torch/onnx/_internal/exporter/_registration.py [hunk 1]
```diff
@@ -61,18 +62,9 @@ def _get_overload(qualified_name: str) -> torch._ops.OpOverload | None:
     if namespace == "math":
         return getattr(math, op_name)
     if namespace == "torchvision":
-        try:
-            import torchvision.ops  # type: ignore[import-untyped]
-        except ImportError:
+        if importlib.util.find_spec("torchvision.ops") is None:
```
Comment (justinchuby) [inline torch/onnx/_internal/exporter/_registration.py:65]:
```suggestion
        if importlib.util.find_spec("torchvision") is None:
```

thanks for this change

**Comment by titaiwangms** [comment]
@justinchuby I tested this locally, and it actually fixes the bug. We do have [torchvision::nms](https://github.com/microsoft/onnxscript/blob/99cf79fd4ab150e3726b36fb3e9104304e203200/onnxscript/function_libs/torch_lib/ops/vision.py#L17). It's the bug in registration that we wrongly registered it under the function nms itself. In [here](https://github.com/titaiwangms/pytorch/blob/605392bd0630df895efac20a27a6a82e70d9b3de/torch/onnx/_internal/exporter/_registration.py#L70), it will register torchvision::nms symbolic function under nms function itself because that's what we get from `return getattr(torchvision.ops, op_name)`. In fact, torchvision ops are accessible the same way as aten ops it seems.

**Comment by titaiwangms** [comment]
@Bludator Do you mind adding your repro as a test here: https://github.com/titaiwangms/pytorch/blob/main/test/onnx/exporter/test_small_models_e2e.py

**Comment by justinchuby** [comment]
> @justinchuby I tested this locally, and it actually fixes the bug. We do have [torchvision::nms](https://github.com/microsoft/onnxscript/blob/99cf79fd4ab150e3726b36fb3e9104304e203200/onnxscript/function_libs/torch_lib/ops/vision.py#L17). It's the bug in registration that we wrongly registered it under the function nms itself. In [here](https://github.com/titaiwangms/pytorch/blob/605392bd0630df895efac20a27a6a82e70d9b3de/torch/onnx/_internal/exporter/_registration.py#L70), it will register torchvision::nms symbolic function under nms function itself because that's what we get from `return getattr(torchvision.ops, op_name)`. In fact, torchvision ops are accessible the same way as aten ops it seems.

thank you!

**Review by justinchuby — COMMENTED** [review]
LGTM if torch.ops.torchvision.* is defined

**Comment by Bludator** [comment]
> @Bludator Do you mind adding your repro as a test here: https://github.com/titaiwangms/pytorch/blob/main/test/onnx/exporter/test_small_models_e2e.py

Is the torchvision installed in the test environment?

**Comment by titaiwangms** [comment]
> > @Bludator Do you mind adding your repro as a test here: https://github.com/titaiwangms/pytorch/blob/main/test/onnx/exporter/test_small_models_e2e.py
> 
> Is the torchvision installed in the test environment?

I believe so. We have torchvision [tests](https://github.com/pytorch/pytorch/blob/45ed7c13fafcc16687e4b80c8ab13d2e2834e954/test/onnx/test_pytorch_onnx_onnxruntime.py#L8799) for legacy torchscript exporter.

File: test/onnx/exporter/test_small_models_e2e.py [hunk 2]
```diff
@@ -125,6 +127,27 @@ def false_fn(x):
...
+        args = (
+            (
+                torch.rand(20, 4, dtype=torch.float),
+                torch.rand(20, dtype=torch.float),
+                torch.randint(0, 2, (20,), dtype=torch.float),
+                0,
+            ),
+            3,
+        )
+        torch.onnx.export(VisionModel(), args, dynamo=True)
```
Comment (titaiwangms) [inline test/onnx/exporter/test_small_models_e2e.py:149]:
```suggestion
        onnx_program = torch.onnx.export(VisionModel(), args, dynamo=True)
        onnx_testing.assert_onnx_program(onnx_program)
```

**Review by titaiwangms — APPROVED** [review]
Thanks!

**Comment by Bludator** [comment]
Now it should work, I had broken environment so I didn't properly test it.
@pytorchbot merge

**Pull request closed without merging**